lib/ssh/src/ssh_system_sup.erl

```diff
@@ -53,15 +53,26 @@
 
 start_system(Role, Address0, Options) ->
     case find_system_sup(Role, Address0) of
-        {ok,{SysPid,Address}} ->
+        {ok,{SysPid,Address}} when Role =:= server ->
             restart_acceptor(SysPid, Address, Options);
+        {ok,{SysPid,_Address}} ->
+            {ok, SysPid};
         {error,not_found} ->
-            supervisor:start_child(sup(Role),
+            case supervisor:start_child(sup(Role),
                                    #{id       => {?MODULE,Address0},
                                      start    => {?MODULE, start_link, [Role, Address0, Options]},
                                      restart  => temporary,
                                      type     => supervisor
                                     })
+            of
+                {ok, SysPid} ->
+                    {ok, SysPid};
+                {error, {already_started, SysPid}} ->
+                    %% There was other connection that created the supervisor while
+                    %% this process was trying to create it as well
+                    {ok, SysPid};
```

Comment:
I totally missed that address had the local port and not the destination port. Good catch. Given that I think I am back to the drawing board with the RACE2. 

I haven't actually tried just to run the adjustments for RACE1 and RACE3 without RACE2. I can try that but it will take me quite some time to get results I think. One thing is for certain, we were getting `already_started` errors from the ssh subsystem and we no longer get those after these changes. And even though I don't have the logs any longer I am quite certain that the already_started error came from start_system.

Shall I remove the changes regarding RACE2 from this PR, so we can proceed with the fixes for RACE1 and RACE3?